Sources/SE0000_KeyPathReflection/Reflection.swift

```diff
@@ -54,12 +57,12 @@ public enum Reflection {
   ///            for this value.
   public static func allNamedKeyPaths<T>(
     for type: T.Type
-  ) -> [(name: String, keyPath: PartialKeyPath<T>)] {
+  ) -> [(name: StaticString, keyPath: PartialKeyPath<T>)] {
```

Comment:
I'm curious why you chose to go with `StaticString` here instead of just using `String`. I get the fact that you don't want to copy field names, but `String` is the currency type in Swift and all the ergonomic conformances are already on `String`. I notice that you had to even add the `Equatable` conformance to `StaticString` to compare field names. Considering we can't back deploy conformances at the moment and that comparing field names seems common enough, doesn't it make sense to stick to `String` even though we need to copy the string contents?

Reply:
Reverted for now.

I agree that using the currency type is better, but I just wanted to make sure this API is designed for efficiency.  In some use cases (e.g. ML use cases I was thinking of) this may be called frequently and repeatedly on large nested structs, and it would be unfortunate if it allocates and copies a bunch of strings every time.  Hopefully when this lands in the standard library we will be able to create `String`s efficiently without copying.

Reply:
Ah I see your point. Yeah hopefully we can workaround this for the stdlib.